ooniprobe/AppDelegate.mm

```diff
@@ -220,7 +220,12 @@ -(void)openURIschemeScreen:(NSDictionary*)parameters{
             [rvc setTestArguments:[parameters objectForKey:@"ta"]];
         if ([parameters objectForKey:@"td"])
             [rvc setTestDescription:[parameters objectForKey:@"td"]];
-        [self.window.rootViewController presentViewController:nvc animated:YES completion:nil];
+        if (self.window.rootViewController.view.window != nil)
+            [self.window.rootViewController presentViewController:nvc animated:YES completion:nil];
+        else {
+            //view is not in the window hierarchy
+            [[NSNotificationCenter defaultCenter] postNotificationName:@"reloadTest" object:nil userInfo:parameters];
+        }
```

Comment:
Which is the case where you have already created a view and you're just recycling it? The second?

Reply:
yes

Reply:
Okay, but why would the `.window` be `!= nil` in that case?

Reply:
In this case view.window is the ViewController of the main app (the one with run tests etc) not the overlay view.
It's nil when is covered. Or at least it's what I learnt from stackoverflow

Reply:
ack!

Reply:
Can you maybe add a comment saying that and referencing the stack-overflow link?

Reply:
https://stackoverflow.com/questions/26022756/warning-attempt-to-present-on-whose-view-is-not-in-the-window-hierarchy-s

https://stackoverflow.com/questions/30941424/is-myviewcontroller-in-the-window-hierarchy

Reply:
I meant, a comment _in the code_

Reply:
Like here: https://github.com/TheTorProject/ooniprobe-ios/pull/141/files#diff-eff536c28a49eda653581c9cd5605e23R223

Reply:
I don't like putting links in the code, expecially from random stackoverflow answers and when what the code is doing is so clear. 
Can I just explain when the view is nill?